Library/Homebrew/cmd/gist-logs.rb

```diff
@@ -115,15 +115,8 @@ def http
   end
 
   def make_request(path, data, auth)
-    headers = {
-      "User-Agent"   => HOMEBREW_USER_AGENT,
-      "Accept"       => "application/vnd.github.v3+json",
-      "Content-Type" => "application/json"
-    }
-
-    if auth == :AUTH_TOKEN || (auth.nil? && HOMEBREW_GITHUB_API_TOKEN)
-      headers["Authorization"] = "token #{HOMEBREW_GITHUB_API_TOKEN}"
-    end
+    headers = GitHub.api_headers
+    headers["Content-Type"] = "application/json"
```

Comment:
One thing we have failed to handled for long time is we didn't check whether the token has gist write access. It may be a good chance to address this. Your call?